src/holdings/holdings.py

```diff
@@ -17,6 +17,7 @@ class Holdings:
     def __init__(self):
         """Constructor"""
         self.data: Dict[UUID, Dict[date, Decimal]] = {}
+        self.day_amounts: Dict[UUID, Dict[date, Decimal]] = {}
 
```

Comment:
How about calculating minimum and maximun while adding holdings so it wont make a call for that calculation each time? Just a suggestion though, for performance.